### PR TITLE
chore: updates and scripts

### PR DESCRIPTION
**Pull request by dominikg**
scripts for prettier and type checking
dependency updates

1 test failure for const-namespace where it turns it into a let instead of a const.

### CODE REVIEW

**Comment by dominikg** [comment]
would you consider commiting the generated types and ensure they are in sync  with a script before release?

eg https://github.com/dominikg/tsconfck/blob/b436363b353d76a55947a672593344a57b10ab62/.github/workflows/release.yml#L50

otherwise it may be good to call the cli as first step for check:types to ensure they are in sync

**Comment by dominikg** [comment]
reverting back to ~5.0.4 lets the test pass, not sure what exactly causes it, but this reads like it could affect dts-buddy https://devblogs.microsoft.com/typescript/announcing-typescript-5-1/#reduced-calls-into-scanner-for-jsdoc-parsing

File: package.json [hunk 1]
```diff
@@ -3,26 +3,29 @@
...
+		"@types/node": "^20.5.1",
+		"prettier": "^3.0.2",
 		"uvu": "^0.5.6"
 	},
 	"scripts": {
 		"prepublishOnly": "node src/cli.js",
-		"test": "node test/test.js"
+		"test": "node test/test.js",
+		"check:types": "tsc --noEmit --emitDeclarationOnly false",
+		"check:format": "prettier --ignore-path .gitignore --ignore-path .prettierignore --cache --check .",
```
[the remark this answers is not in the document]
Comment (dominikg) [reply package.json]: check: allows for more flexible composition and adding more stuff like publint later, see https://github.com/sveltejs/vite-plugin-svelte/blob/main/package.json 

changed it to lint and check anywasy. :man_shrugging:

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -32,4 +32,12 @@ jobs:
           node-version: ${{ matrix.node-version }}
           cache: pnpm
       - run: pnpm install --frozen-lockfile
-      - run: pnpm test
+      - name: lint
+        if: ${{ !cancelled() }}
```
Comment (benmccann) [inline .github/workflows/ci.yml:36]: do we really need `if: ${{ !cancelled() }}` everywhere?

Comment (dominikg) [reply .github/workflows/ci.yml:36]: this allows a single run to output all failed commands

Comment (benmccann) [reply .github/workflows/ci.yml:36]:
```suggestion
        if: ${{ !cancelled() }} # allows checks to run if one fails so we can put them all in the same job
```

File: .editorconfig [hunk 3]
```diff
@@ -0,0 +1,12 @@
+root = true
```
[the remark this answers is not in the document]
Comment (dominikg) [reply .editorconfig:1]: this helps people who don't have it set up and even those who do. prettier is run on save mostly (or just when calling the script). .editorconfig otoh usually is applied to the active editor while you type, eg insert the right indentation chars.

**Review by benmccann — APPROVED** [review]

**Comment by Rich-Harris** [comment]
closing in favour of #45

**Pull request closed without merging**